code/modules/events/wizard/rpgloot.dm

```diff
@@ -36,28 +19,97 @@
 /obj/item/upgradescroll
-	name = "Item Fortification Scroll"
+	name = "item fortification scroll"
 	desc = "Somehow, this piece of paper can be applied to items to make them \"better\". Apparently there's a risk of losing the item if it's already \"too good\". <i>This all feels so arbitrary...</i>"
 	icon = 'icons/obj/wizard.dmi'
 	icon_state = "scroll"
 	w_class = WEIGHT_CLASS_TINY
 
+	var/upgrade_amount = 1
+	var/can_backfire = TRUE
+	var/one_use = TRUE
+
 /obj/item/upgradescroll/afterattack(obj/item/target, mob/user , proximity)
 	if(!proximity || !istype(target))
 		return
-	var/quality = target.force - initial(target.force)
-	if(quality > 9 && prob((quality - 9)*10))
-		to_chat(user, "<span class='danger'>[target] catches fire!</span>")
-		if(target.resistance_flags & (LAVA_PROOF|FIRE_PROOF))
-			target.resistance_flags &= ~(LAVA_PROOF|FIRE_PROOF)
-			target.resistance_flags |= FLAMMABLE
-		target.fire_act()
+
+	var/datum/rpg_loot/rpg_loot_datum = target.rpg_loot
+	if(!istype(rpg_loot_datum))
+		new rpg_loot_datum(target)
+
+	var/quality = rpg_loot_datum.quality
+
+	if(can_backfire && (quality > 9 && prob((quality - 9)*10)))
+		to_chat(user, "<span class='danger'>[target] violently glows blue for a while, then evaporates.</span>")
+		qdel(target)
+	else
+		to_chat(user, "<span class='notice'>[target] glows blue and seems vaguely \"better\"!</span>")
+		rpg_loot_datum.modify(upgrade_amount)
+
+	if(one_use)
 		qdel(src)
-		return
-	target.force 		+= 1
-	target.throwforce	+= 1
-	for(var/value in target.armor)
-		target.armor[value] += 1
-	to_chat(user, "<span class='notice'>[target] glows blue and seems vaguely \"better\"!</span>")
-	qdel(src)
+
+/obj/item/upgradescroll/unlimited
+	name = "unlimited foolproof item fortification scroll"
+	desc = "Somehow, this piece of paper can be applied to items to make them \"better\". This scroll is made from the tongues of dead paper wizards, and can be used an unlimited number of times, with no drawbacks."
+	one_use = FALSE
+	can_backfire = FALSE
+
+/datum/rpg_loot
+	var/positive_prefix = "okay"
+	var/negative_prefix = "weak"
+	var/suffix = "something profound"
+	var/quality = 0
+
+	var/obj/item/attached
+	var/original_name
+
+/datum/rpg_loot/New(attached_item=null)
+	attached = attached_item
+
+	randomise()
+
+/datum/rpg_loot/Destroy()
+	attached = null
+
+/datum/rpg_loot/proc/randomise()
+	var/list/prefixespositive 	= list("greater", "major", "blessed", "superior", "enpowered", "honed", "true", "glorious", "robust")
+	var/list/prefixesnegative 	= list("lesser", "minor", "blighted", "inferior", "enfeebled", "rusted", "unsteady", "tragic", "gimped")
+	var/list/suffixes			= list("orc slaying", "elf slaying", "corgi slaying", "strength", "dexterity", "constitution", "intelligence", "wisdom", "charisma", "the forest", "the hills", "the plains", "the sea", "the sun", "the moon", "the void", "the world", "the fool", "many secrets", "many tales", "many colors", "rending", "sundering", "the night", "the day")
+
```

Comment:
If I'm reading this right, `randomise()` fires as part of `New`, which, in the case of the RPG Loot event, happens for every item in the world, and creates new lists for every instance, for its own call of `randomise()` .

The code this replaces creates the lists only once, in `rpgloot/start()`.

Is that a bit resource intensive? Somebody please correct me if I'm wrong, but this might be a time for using static or a global.